### PR TITLE
DEPL-11598 Reload types fails when using Jenkins folder credentials

### PR DESCRIPTION
**Pull request by aloke-xebia**

### CODE REVIEW

File: src/main/java/com/xebialabs/deployit/ci/Credential.java [hunk 1]
```diff
@@ -241,6 +242,13 @@ public static StandardUsernamePasswordCredentials lookupSystemCredentials(String
         );
     }
 
+    public static StandardUsernamePasswordCredentials lookupSystemCredentials(String credentialsId, Item itemGroup) {
+        return CredentialsMatchers.firstOrNull(
+                lookupCredentials(StandardUsernamePasswordCredentials.class, itemGroup, ACL.SYSTEM,
```
Comment (sameerarora) [inline src/main/java/com/xebialabs/deployit/ci/Credential.java]: This code is almost the same as the other lookupSystemCredentials , can we make the call at one place and delegate to that from another. I think ItemGroup and Jenkins should be in the same hierarchy.

File: src/main/java/com/xebialabs/deployit/ci/RepositoryUtils.java [hunk 2]
```diff
@@ -46,6 +47,19 @@ public static Credential retrieveOverridingCredentialFromProject(AbstractProject
         DeployitNotifier notifier = retrieveDeployitNotifierFromProject(project);
         if (null != notifier) {
             overridingCredential = notifier.getOverridingCredential();
+            if ((null == overridingCredential.getUsername() || "".equals(overridingCredential.getUsername()))
```
Comment (sameerarora) [inline src/main/java/com/xebialabs/deployit/ci/RepositoryUtils.java]: StringUtils.isEmpty(username)

File: src/main/java/com/xebialabs/deployit/ci/RepositoryUtils.java [hunk 3]
```diff
@@ -46,6 +47,19 @@ public static Credential retrieveOverridingCredentialFromProject(AbstractProject
...
+                    && null != overridingCredential.getCredentialsId()) {
+                DeployitNotifier.DeployitDescriptor descriptor = (DeployitNotifier.DeployitDescriptor) notifier.getDescriptor();
```
Comment (sameerarora) [inline src/main/java/com/xebialabs/deployit/ci/RepositoryUtils.java]: replace qualified imports with DeployitDescriptor

File: src/test/java/com/xebialabs/deployit/ci/RepositoryUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        DeployitNotifier.DeployitDescriptor descriptor = new DeployitNotifier.DeployitDescriptor();
+
+        DeployitNotifier notifierSpy = Mockito.spy(new DeployitNotifier("AdminGlobal1", "app1", null, null, null, null, false, null, overridingCredentialProvided));
+        doReturn(descriptor).when(notifierSpy).getDescriptor();
+        doReturn(overridingCredentialProvided).when(notifierSpy).getOverridingCredential();
+
+        FreeStyleProject freeStyleProjectSpy = Mockito.spy(new FreeStyleProject(f, "folder1/proj1"));
+        freeStyleProjectSpy.addPublisher(notifierSpy);
+
+        DescribableList<Publisher, Descriptor<Publisher>> publisherListMock = Mockito.mock(DescribableList.class);
```
Comment (sameerarora) [inline src/test/java/com/xebialabs/deployit/ci/RepositoryUtilsTest.java]: please add a static import for Mockito.spy

File: src/main/java/com/xebialabs/deployit/ci/Credential.java (hunk 1 above)
Comment (aloke-xebia) [inline src/main/java/com/xebialabs/deployit/ci/Credential.java]: Can't be done. One method's for itemGroup, while the other one's for item.

File: src/main/java/com/xebialabs/deployit/ci/RepositoryUtils.java (hunk 3 above)
Comment (aloke-xebia) [inline src/main/java/com/xebialabs/deployit/ci/RepositoryUtils.java]: Done

File: src/main/java/com/xebialabs/deployit/ci/RepositoryUtils.java (hunk 2 above)
Comment (aloke-xebia) [inline src/main/java/com/xebialabs/deployit/ci/RepositoryUtils.java]: Done

File: src/test/java/com/xebialabs/deployit/ci/RepositoryUtilsTest.java (hunk 4 above)
Comment (aloke-xebia) [inline src/test/java/com/xebialabs/deployit/ci/RepositoryUtilsTest.java]: Done

**Pull request merged**